Tests/GoldEtlCsvTests.cpp

```diff
@@ -102,21 +102,42 @@ class Tests : public ::testing::Test, TestArgs {
                 if (testCsv.headerColumnIndex_[h] != SIZE_MAX && goldCsv.headerColumnIndex_[h] != SIZE_MAX) {
                     char const* a = testCsv.cols_[testCsv.headerColumnIndex_[h]];
                     char const* b = goldCsv.cols_[goldCsv.headerColumnIndex_[h]];
-                    if (_stricmp(a, b) != 0) {
-                        if (rowOk) {
-                            rowOk = false;
-                            printf("GOLD = %ls\n", goldCsv_.c_str());
-                            printf("TEST = %ls\n", testCsv_.c_str());
-                            AddTestFailure(__FILE__, __LINE__, "Difference on line: %zu", testCsv.line_);
-                            printf("    COLUMN                    TEST VALUE                            GOLD VALUE\n");
+                    if (_stricmp(a, b) == 0) {
+                        continue;
+                    }
+
+                    // Floating point may be inconsistently rounded by printf accross different platforms.
+                    // Do a rounding check by ensuring the difference between the two numebrs are less than 
+                    // the final digit +-1.001
+
+                    // Doubles should be good for 15 digits of precision.
+
+                    double testNumber = strtod(a, nullptr);
+                    double goldNumber = strtod(b, nullptr);
+                    const char* testDecimalAddr = strchr(a, '.');
+                    const char* goldDecimalAddr = strchr(b, '.');
+                    if (testNumber != 0.0 && goldNumber != 0.0 && testDecimalAddr != NULL && goldDecimalAddr != NULL) {
+                        size_t testDecimalNumbersCount = (a + strlen(a)) - testDecimalAddr - 1;
+                        size_t goldDecimalNumbersCount = (b + strlen(b)) - goldDecimalAddr - 1;
+                        double difference = (testNumber * pow(10.0, testDecimalNumbersCount)) - (goldNumber * pow(10.0, goldDecimalNumbersCount));
+                        if (difference > -1.001 && difference < 1.001) {

```

Comment:
You shouldn't multiply testNumber and goldNumber by different factors here, so check that testDecimalNumbersCount==goldDecimalNumbersCount first.  It may be better to just scale the '1' instead (?).

Reply:
Yeah, that makes sense. I think scaling the 1 means scaling 1.001/-1.001? Can do that.

Reply:
Ended up comparing the decimal magnitude between the two numbers instead. It still has the assumption that the spreadsheet will have the same number of decimal places, but I think that's an reasonable expectation and the code is slightly more intuitive if the multiplication happens to the scanned numbers rather than 1.001.